### PR TITLE
Start Arguments Bug Fix

### PR DESCRIPTION
**Pull request by nicktate**
added back in support to disable standard docker logs. also fixed issue with start arguments not converting function values into their executed values

@normanjoyner @jeremykross

### CODE REVIEW

**Comment by jeremykross** [comment]
👍

**Comment by jeremykross** [comment]
👍

File: engines/docker-engine.js [hunk 1]
```diff
@@ -64,15 +65,28 @@ class DockerEngine extends Engine {
...
 
-        return _.merge.apply(null, _.map(options, (v,k) => (mapping[k] ? mapping[k](v): null)));
+        mapping = _.merge.apply(null, _.map(options, (v,k) => (mapping[k] ? mapping[k](v): null)));
+
+		// merge in additional start aguments to the options mapping
+		// start argument values may have been set as functions
+		// expecting to be invoked with the existing options
+		let start_args = flat.flatten(options.start_args);
+		start_args = _.reduce(start_args, (accumulator, value, key) => {
+			accumulator[key] = Utils.safeJsonParse(typeof value === 'function' ? value(mapping) : value)
```
Comment (normanjoyner) [inline engines/docker-engine.js]: semicolon

File: engines/docker-engine.js [hunk 2]
```diff
@@ -64,15 +65,28 @@ class DockerEngine extends Engine {
...
+                    'ExposedPorts': exposedPorts
                 }
             }
 
         };
 
-        return _.merge.apply(null, _.map(options, (v,k) => (mapping[k] ? mapping[k](v): null)));
+        mapping = _.merge.apply(null, _.map(options, (v,k) => (mapping[k] ? mapping[k](v): null)));
+
+		// merge in additional start aguments to the options mapping
```
Comment (normanjoyner) [inline engines/docker-engine.js]: spacing

**Pull request merged**